src/Storages/System/StorageSystemDashboards.cpp

```diff
@@ -0,0 +1,226 @@
+WHERE event_date >= toDate(now() - {seconds:UInt32}) AND event_time >= now() - {seconds:UInt32}
+GROUP BY t
+ORDER BY t WITH FILL STEP {rounding:UInt32}
+)EOQ") }
+        },
+        {
+            { "dashboard", "overview" },
+            { "title", "Memory (tracked)" },
+            { "query", trim(R"EOQ(
+SELECT toStartOfInterval(event_time, INTERVAL {rounding:UInt32} SECOND)::INT AS t, avg(CurrentMetric_MemoryTracking)
+FROM system.metric_log
+WHERE event_date >= toDate(now() - {seconds:UInt32}) AND event_time >= now() - {seconds:UInt32}
+GROUP BY t
+ORDER BY t WITH FILL STEP {rounding:UInt32}
+)EOQ") }
+        },
+        {
+            { "dashboard", "overview" },
+            { "title", "Load Average (15 minutes)" },
+            { "query", trim(R"EOQ(
+SELECT toStartOfInterval(event_time, INTERVAL {rounding:UInt32} SECOND)::INT AS t, avg(value)
+FROM system.asynchronous_metric_log
+WHERE event_date >= toDate(now() - {seconds:UInt32}) AND event_time >= now() - {seconds:UInt32} AND metric = 'LoadAverage15'
+GROUP BY t
+ORDER BY t WITH FILL STEP {rounding:UInt32}
+)EOQ") }
+        },
+        {
+            { "dashboard", "overview" },
+            { "title", "Selected Rows/second" },
+            { "query", trim(R"EOQ(
+SELECT toStartOfInterval(event_time, INTERVAL {rounding:UInt32} SECOND)::INT AS t, avg(ProfileEvent_SelectedRows)
+FROM system.metric_log
+WHERE event_date >= toDate(now() - {seconds:UInt32}) AND event_time >= now() - {seconds:UInt32}
+GROUP BY t
+ORDER BY t WITH FILL STEP {rounding:UInt32}
+)EOQ") }
+        },
+        {
+            { "dashboard", "overview" },
+            { "title", "Inserted Rows/second" },
+            { "query", trim(R"EOQ(
+SELECT toStartOfInterval(event_time, INTERVAL {rounding:UInt32} SECOND)::INT AS t, avg(ProfileEvent_InsertedRows)
+FROM system.metric_log
+WHERE event_date >= toDate(now() - {seconds:UInt32}) AND event_time >= now() - {seconds:UInt32}
+GROUP BY t
+ORDER BY t WITH FILL STEP {rounding:UInt32}
+)EOQ") }
+        },
+        {
+            { "dashboard", "overview" },
+            { "title", "Total MergeTree Parts" },
+            { "query", trim(R"EOQ(
+SELECT toStartOfInterval(event_time, INTERVAL {rounding:UInt32} SECOND)::INT AS t, avg(value)
+FROM system.asynchronous_metric_log
+WHERE event_date >= toDate(now() - {seconds:UInt32}) AND event_time >= now() - {seconds:UInt32} AND metric = 'TotalPartsOfMergeTreeTables'
+GROUP BY t
+ORDER BY t WITH FILL STEP {rounding:UInt32}
+)EOQ") }
+        },
+        {
+            { "dashboard", "overview" },
+            { "title", "Max Parts For Partition" },
+            { "query", trim(R"EOQ(
+SELECT toStartOfInterval(event_time, INTERVAL {rounding:UInt32} SECOND)::INT AS t, max(value)
+FROM system.asynchronous_metric_log
+WHERE event_date >= toDate(now() - {seconds:UInt32}) AND event_time >= now() - {seconds:UInt32} AND metric = 'MaxPartCountForPartition'
+GROUP BY t
+ORDER BY t WITH FILL STEP {rounding:UInt32}
+)EOQ") }
+        }
```

Comment:
Interesting idea. What do you think about moving this into `config.xml`? That way, user can customize it.

Reply:
Yes, interesting. I was thinking about making another writable table `system.custom_dashboards` to make it changeable even in runtime... but it might be an overkill. I like your idea.

Reply:
I wonder how we migrate to this... Because releasing the version with charts read from `config.xml` will make `system.dashboards` become an empty table unless `config.xml` is also updated in sync...

Reply:
We can add this predefined charts only if nothing had been configured by user in config. That way it will be compatible.

And also it will have one more bonus - it will make this charts available even after upgrade, since not all users sync config.xml during upgrade (they ship it with ansible or similar tools) and without defaults in code they will see nothing.

Reply:
This way we need to have the same charts both in some .xml and in .cpp file (unless we use some magic to include .xml file contents in our binary, as It is done e.g. for `dashboard.html`).

Also if we do this transition it would be hard to add new charts just by releasing clickhouse binary. If one NOT define it's own charts, then they will be auto-updated. If one indeed define charts, then they should look for updates and deploy newer charts with the release. So it is a different deploy procedures w/ and w/o custom charts. I do not like this.

Maybe the solution is just to **combine** built-in charts in .cpp with charts in .xml instead of **replacing**?

Reply:
>This way we need to have the same charts both in some .xml and in .cpp file (unless we use some magic to include .xml file contents in our binary, as It is done e.g. for dashboard.html).

Not really, I mean that if something is configured by user, we just consider that this is all what he wants and do not show builtins.

>Also if we do this transition it would be hard to add new charts just by releasing clickhouse binary. If one NOT define it's own charts, then they will be auto-updated. If one indeed define charts, then they should look for updates and deploy newer charts with the release. So it is a different deploy procedures w/ and w/o custom charts. I do not like this.

If someone changed them, then likely he knows what he is doing (I guess <1% will do this).
For example I would likely to configure them, plus, I may not be interested in some default charts either.

>Maybe the solution is just to combine built-in charts in .cpp with charts in .xml instead of replacing?

That way you cannot remove some charts...
Personally I, want this ability.

Reply:
Something like this - https://github.com/ClickHouse/ClickHouse/pull/65555